crates/oxc_transformer/src/commonjs/utils/export.rs

```diff
@@ -0,0 +1,278 @@
+/// ```js
+/// exports.default = foo
+/// ```
+pub fn create_default_exports<'a>(
+    declaration: Expression<'a>,
+    builder: &'a AstBuilder,
+) -> Statement<'a> {
+    if declaration.is_identifier_reference() {
+        builder.statement_expression(
+            SPAN,
+            create_exports(
+                builder.module_export_name_identifier_name(SPAN, "default"),
+                declaration,
+                builder,
+            ),
+        )
+    } else {
+        create_exports_with_assignment(
+            builder.vec1((
+                "default",
+                builder.module_export_name_identifier_name(SPAN, "_default"),
+                declaration,
+            )),
+            VariableDeclarationKind::Var,
+            builder,
+        )
+    }
+}
+
+pub fn create_declared_named_exports<'a>(
+    declaration: Declaration<'a>,
+    builder: &'a AstBuilder,
+) -> Vec<'a, Statement<'a>> {
+    match declaration {
+        Declaration::VariableDeclaration(decls) => {
+            let mut result = builder.vec();
+            for decl in decls.declarations.iter() {
+                match &decl.id.kind {
+                    BindingPatternKind::BindingIdentifier(id) => {
+                        result.push(builder.statement_expression(
+                            SPAN,
+                            create_exports(
+                                builder.module_export_name_identifier_name(SPAN, id.name.as_str()),
+                                match &decl.init {
+                                    Some(init) => init.clone_in(builder.allocator),
+                                    None => builder.void_0(),
+                                },
+                                builder,
+                            ),
+                        ))
+                    }
+                    _ => unreachable!(),
+                }
+            }
+            result
+        }
+        Declaration::FunctionDeclaration(decls) => {
+            let mut result = builder.vec();
+            // 1. append the function declaration without export
+            result.push(builder.statement_expression(
+                SPAN,
+                builder.expression_from_function(decls.clone_in(builder.allocator)),
+            ));
+            // 2. append the export statement
+            let identifier = &decls.id;
+            match identifier {
+                Some(id) => result.push(builder.statement_expression(
+                    SPAN,
+                    create_exports(
+                        builder.module_export_name_identifier_reference(SPAN, id.name.as_str()),
+                        builder.expression_identifier_reference(SPAN, id.name.as_str()),
+                        builder,
+                    ),
+                )),
+                None => unreachable!(),
+            }
+            result
+        }
+        Declaration::ClassDeclaration(decls) => {
+            let mut result = builder.vec();
```

Comment:
Here is better using `AstBuilder::vec_with_capacity`.

Other, here we need to pre-assign class idenitfier to `exports`. Like 
``` .ts
export class a {}
export class b {}
```

It should be transformed to 
``` .ts
exports.b = exports.a = void 0;
class a {}
exports.a = a;
class b {}
exports.b = b;
```